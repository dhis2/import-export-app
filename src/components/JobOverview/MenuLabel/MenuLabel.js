import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import { jsDateToString } from '../../../utils/helper.js'
import styles from './MenuLabel.module.css'

const MenuLabel = ({ task }) => {
    return (
        <div
            className={styles.container}
            data-test={`job-overview-menu-label-${task.id}`}
        >
            <div className={styles.status}>
                <span className={styles.filename}>
                    {task.jobDetails.files[0].name}
                </span>
                <br />
                <span>
                    {task.completed
                        ? i18n.t('Completed')
                        : i18n.t('In progress')}{' '}
                </span>
            </div>
            <div className={styles.date}>{jsDateToString(task.created)}</div>
        </div>
    )
}

MenuLabel.propTypes = {
    task: PropTypes.shape({
        completed: PropTypes.bool.isRequired,
        created: PropTypes.instanceOf(Date).isRequired,
        id: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
            .isRequired,
        jobDetails: PropTypes.object.isRequired,
    }).isRequired,
}

export { MenuLabel }
