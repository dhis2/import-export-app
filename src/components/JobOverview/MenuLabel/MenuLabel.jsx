import i18n from '@dhis2/d2-i18n'
import { Tag } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { jsDateToString } from '../../../utils/helper.js'
import styles from './MenuLabel.module.css'

const MenuLabel = ({ task }) => {
    return (
        <div
            className={styles.container}
            data-test={`job-overview-menu-label-${task.id}`}
        >
            <div className={styles.filename}>
                {task.jobDetails.files[0].name}
            </div>
            <div className={styles.subtitle}>
                {task.completed ? (
                    <Tag positive>{i18n.t('Completed')}</Tag>
                ) : (
                    <Tag>{i18n.t('In progress')}</Tag>
                )}{' '}
                <p className={styles.date}>{jsDateToString(task.created)}</p>
            </div>
        </div>
    )
}

MenuLabel.propTypes = {
    task: PropTypes.shape({
        completed: PropTypes.bool.isRequired,
        created: PropTypes.instanceOf(Date).isRequired,
        id: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
            .isRequired,
        jobDetails: PropTypes.object.isRequired,
    }).isRequired,
}

export { MenuLabel }
