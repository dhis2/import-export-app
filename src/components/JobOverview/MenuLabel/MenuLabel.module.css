.container {
    display: flex;
    flex-direction: column;
    gap: var(--spacers-dp8);
}

.filename {
    overflow: hidden;
    text-overflow: ellipsis;
}

.subtitle {
    display: flex;
    gap: var(--spacers-dp8);
    align-items: center;
}

.date {
    font-size: 13px;
    color: var(--colors-grey700);
    margin: 0;
}

@media (min-width: 1200px) {
    .filename {
        max-width: 290px;
    }
}
