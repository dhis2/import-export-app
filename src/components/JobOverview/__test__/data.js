export const tasksState = {
    data: {
        EcJpZjEcMfI: {
            id: 'EcJpZjEcMfI',
            level: 'INFO',
            created: new Date('2020-02-12T13:00:23.350'),
            lastUpdated: new Date('2020-02-12T13:00:23.766'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Initiated dataValueImport',
                    date: new Date('2020-02-12T12:00:23.230'),
                },
                {
                    id: 'LPCnEwppbqy',
                    level: 'INFO',
                    text: 'Process started',
                    date: new Date('2020-02-12T12:00:23.232'),
                },
                {
                    id: 'quL1PHwVPg1',
                    level: 'INFO',
                    text: 'Importing data values',
                    date: new Date('2020-02-12T12:00:23.498'),
                },
                {
                    id: 'fn5gnxqYk1J',
                    level: 'INFO',
                    text: 'Import done',
                    date: new Date('2020-02-12T12:00:23.499'),
                },
            ],
            summary: {
                responseType: 'ImportSummary',
                status: 'SUCCESS',
                importOptions: {
                    idSchemes: {},
                    dryRun: true,
                    preheatCache: true,
                    async: true,
                    importStrategy: 'NEW_AND_UPDATES',
                    mergeMode: 'REPLACE',
                    reportMode: 'FULL',
                    skipExistingCheck: false,
                    sharing: false,
                    skipNotifications: false,
                    skipAudit: false,
                    datasetAllowsPeriods: false,
                    strictPeriods: false,
                    strictDataElements: false,
                    strictCategoryOptionCombos: false,
                    strictAttributeOptionCombos: false,
                    strictOrganisationUnits: false,
                    requireCategoryOptionCombo: false,
                    requireAttributeOptionCombo: false,
                    skipPatternValidation: false,
                    ignoreEmptyCollection: false,
                    force: false,
                    firstRowIsHeader: true,
                    skipLastUpdated: false,
                },
                description: 'Import process completed successfully',
                importCount: {
                    imported: 0,
                    updated: 0,
                    ignored: 0,
                    deleted: 0,
                },
                dataSetComplete: 'false',
            },
            error: false,
            importType: 'DATAVALUE_IMPORT',
            jobDetails: {
                files: [new File([], 'data1.json')],
                format: {
                    value: 'json',
                    label: 'JSON',
                },
                dryRun: true,
                strategy: {
                    value: 'NEW_AND_UPDATES',
                    label: 'New and updates',
                },
                preheatCache: true,
                skipAudit: false,
                dataElementIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                orgUnitIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                idScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                skipExistingCheck: false,
                firstRowIsHeader: false,
            },
        },
        SN0RKLRMWBY: {
            id: 'SN0RKLRMWBY',
            level: 'INFO',
            created: new Date('2020-02-12T13:00:59.136'),
            lastUpdated: new Date('2020-02-12T13:00:59.240'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Initiated dataValueImport',
                    date: new Date('2020-02-12T12:00:58.999'),
                },
                {
                    id: 'oE17j9gw54i',
                    level: 'INFO',
                    text: 'Process started',
                    date: new Date('2020-02-12T12:00:59.001'),
                },
                {
                    id: 'SKEyN7hdwyd',
                    level: 'INFO',
                    text: 'Importing data values',
                    date: new Date('2020-02-12T12:00:59.035'),
                },
                {
                    id: 'CO2RDYqaGde',
                    level: 'INFO',
                    text: 'Import done',
                    date: new Date('2020-02-12T12:00:59.148'),
                },
            ],
            summary: {
                responseType: 'ImportSummary',
                status: 'WARNING',
                importOptions: {
                    idSchemes: {},
                    dryRun: true,
                    preheatCache: false,
                    async: true,
                    importStrategy: 'UPDATES',
                    mergeMode: 'REPLACE',
                    reportMode: 'FULL',
                    skipExistingCheck: true,
                    sharing: false,
                    skipNotifications: false,
                    skipAudit: false,
                    datasetAllowsPeriods: false,
                    strictPeriods: false,
                    strictDataElements: false,
                    strictCategoryOptionCombos: false,
                    strictAttributeOptionCombos: false,
                    strictOrganisationUnits: false,
                    requireCategoryOptionCombo: false,
                    requireAttributeOptionCombo: false,
                    skipPatternValidation: false,
                    ignoreEmptyCollection: false,
                    force: false,
                    firstRowIsHeader: true,
                    skipLastUpdated: false,
                },
                description: 'Import process completed successfully',
                importCount: {
                    imported: 0,
                    updated: 0,
                    ignored: 173,
                    deleted: 0,
                },
                conflicts: [
                    {
                        object: 'V6L425pT3A0',
                        value: 'Category option combo not found or not accessible for writing data',
                    },
                    {
                        object: 'Prlt0C1RF0s',
                        value: 'Category option combo not found or not accessible for writing data',
                    },
                    {
                        object: 'hEFKSsPV5et',
                        value: 'Category option combo not found or not accessible for writing data',
                    },
                    {
                        object: 'psbwp3CQEhs',
                        value: 'Category option combo not found or not accessible for writing data',
                    },
                ],
                dataSetComplete: 'false',
            },
            error: false,
            importType: 'DATAVALUE_IMPORT',
            jobDetails: {
                files: [new File([], 'data2.json')],
                format: {
                    value: 'json',
                    label: 'JSON',
                },
                dryRun: true,
                strategy: {
                    value: 'UPDATES',
                    label: 'Updates only',
                },
                preheatCache: false,
                skipAudit: false,
                dataElementIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                orgUnitIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                idScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                skipExistingCheck: true,
                firstRowIsHeader: false,
            },
        },
    },
    event: {
        xW1s5Xo5rWZ: {
            id: 'xW1s5Xo5rWZ',
            level: 'INFO',
            created: new Date('2020-02-12T13:00:28.184'),
            lastUpdated: new Date('2020-02-12T13:00:28.291'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Initiated inMemoryEventImport',
                    date: new Date('2020-02-12T12:00:28.136'),
                },
                {
                    id: 'sso1rmwbq4i',
                    level: 'INFO',
                    text: 'Importing events',
                    date: new Date('2020-02-12T12:00:28.137'),
                },
                {
                    id: 'kRVdpRp3pIy',
                    level: 'INFO',
                    text: 'Import done. Completed in 00:00:00.013.',
                    date: new Date('2020-02-12T12:00:28.154'),
                },
            ],
            summary: {
                responseType: 'ImportSummaries',
                status: 'ERROR',
                imported: 0,
                updated: 0,
                deleted: 0,
                ignored: 1,
                importSummaries: [
                    {
                        responseType: 'ImportSummary',
                        status: 'ERROR',
                        description: 'Event date is required. ',
                        importCount: {
                            imported: 0,
                            updated: 0,
                            ignored: 1,
                            deleted: 0,
                        },
                    },
                ],
                total: 1,
            },
            error: true,
            importType: 'EVENT_IMPORT',
            jobDetails: {
                files: [new File([], 'event1.json')],
                format: {
                    value: 'json',
                    label: 'JSON',
                },
                dryRun: false,
                orgUnitIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
            },
        },
    },
    geojson: {
        1581512432381: {
            id: 1581512432381,
            level: 'INFO',
            created: new Date('2020-02-12T13:00:34.380'),
            lastUpdated: new Date('2020-02-12T13:00:34.380'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Failed to validate job runtime - `GeoJSON import failed.`',
                    date: new Date('2020-02-12T13:00:34.380'),
                },
            ],
            summary: {
                status: 'ERROR',
                stats: {
                    created: 0,
                    updated: 0,
                    deleted: 0,
                    ignored: 0,
                    total: 0,
                },
                typeReports: [
                    {
                        klass: 'org.hisp.dhis.dxf2.geojson.DefaultGeoJsonService',
                        stats: {
                            created: 0,
                            updated: 0,
                            deleted: 0,
                            ignored: 0,
                            total: 0,
                        },
                        objectReports: [
                            {
                                klass: 'org.hisp.dhis.dxf2.geojson.DefaultGeoJsonService',
                                index: 0,
                                errorReports: [
                                    {
                                        message:
                                            'Failed to validate job runtime - `GeoJSON import failed.`',
                                        mainKlass:
                                            'org.hisp.dhis.dxf2.geojson.DefaultGeoJsonService',
                                        errorCode: 'E7010',
                                    },
                                ],
                            },
                        ],
                    },
                ],
            },
            error: true,
            importType: 'GEOJSON_IMPORT',
            jobDetails: {
                files: [new File([], 'geojson1.json')],
                dryRun: true,
            },
        },
    },
    gml: {
        1581512432380: {
            id: 1581512432380,
            level: 'INFO',
            created: new Date('2020-02-12T13:00:32.380'),
            lastUpdated: new Date('2020-02-12T13:00:32.380'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Failed to validate job runtime - `GML import failed: Content is not allowed in prolog. On line 1 column 1.`',
                    date: new Date('2020-02-12T13:00:32.380'),
                },
            ],
            summary: {
                status: 'ERROR',
                stats: {
                    created: 0,
                    updated: 0,
                    deleted: 0,
                    ignored: 0,
                    total: 0,
                },
                typeReports: [
                    {
                        klass: 'org.hisp.dhis.dxf2.gml.DefaultGmlImportService',
                        stats: {
                            created: 0,
                            updated: 0,
                            deleted: 0,
                            ignored: 0,
                            total: 0,
                        },
                        objectReports: [
                            {
                                klass: 'org.hisp.dhis.dxf2.gml.DefaultGmlImportService',
                                index: 0,
                                errorReports: [
                                    {
                                        message:
                                            'Failed to validate job runtime - `GML import failed: Content is not allowed in prolog. On line 1 column 1.`',
                                        mainKlass:
                                            'org.hisp.dhis.dxf2.gml.DefaultGmlImportService',
                                        errorCode: 'E7010',
                                    },
                                ],
                            },
                        ],
                    },
                ],
            },
            error: true,
            importType: 'GML_IMPORT',
            jobDetails: {
                files: [new File([], 'gml1.gml')],
                dryRun: true,
            },
        },
    },
    metadata: {
        zS69mHvSSfj: {
            id: 'zS69mHvSSfj',
            level: 'INFO',
            created: new Date('2020-02-12T13:00:38.990'),
            lastUpdated: new Date('2020-02-12T13:00:39.155'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Initiated metadataImport',
                    date: new Date('2020-02-12T12:00:38.727'),
                },
                {
                    id: 'kltTBpl5Xxy',
                    level: 'INFO',
                    text: '(admin) Import:Start',
                    date: new Date('2020-02-12T12:00:38.751'),
                },
                {
                    id: 'JvqlsZAY0c1',
                    level: 'INFO',
                    text: '(admin) Import:Done took 0.14 seconds',
                    date: new Date('2020-02-12T12:00:38.892'),
                },
            ],
            summary: {
                importParams: {
                    userOverrideMode: 'NONE',
                    importMode: 'COMMIT',
                    identifier: 'UID',
                    preheatMode: 'REFERENCE',
                    importStrategy: 'CREATE_AND_UPDATE',
                    atomicMode: 'ALL',
                    mergeMode: 'MERGE',
                    flushMode: 'AUTO',
                    skipSharing: false,
                    skipTranslation: false,
                    skipValidation: false,
                    metadataSyncImport: false,
                    csvImportClass: 'ORGANISATION_UNIT_GROUP_MEMBERSHIP',
                    firstRowIsHeader: true,
                    username: 'admin',
                },
                status: 'OK',
                stats: {
                    created: 0,
                    updated: 0,
                    deleted: 0,
                    ignored: 0,
                    total: 0,
                },
            },
            error: false,
            importType: 'METADATA_IMPORT',
            jobDetails: {
                files: [new File([], 'metadata1.csv')],
                format: {
                    value: 'csv',
                    label: 'CSV',
                },
                dryRun: false,
                identifier: {
                    value: 'UID',
                    label: 'UID',
                },
                importReportMode: {
                    value: 'ERRORS',
                    label: 'Errors',
                },
                preheatMode: {
                    value: 'REFERENCE',
                    label: 'Reference',
                },
                importStrategy: {
                    value: 'CREATE_AND_UPDATE',
                    label: 'New and updates',
                },
                firstRowIsHeader: true,
                classKey: {
                    value: 'ORGANISATION_UNIT_GROUP_MEMBERSHIP',
                    label: 'ORGANISATION_UNIT_GROUP_MEMBERSHIP',
                },
                atomicMode: {
                    value: 'ALL',
                    label: 'All',
                },
                mergeMode: {
                    value: 'MERGE',
                    label: 'Merge',
                },
                flushMode: {
                    value: 'AUTO',
                    label: 'Auto',
                },
                inclusionStrategy: {
                    value: 'NON_NULL',
                    label: 'Non Null',
                },
                skipSharing: false,
                skipValidation: false,
                isAsync: true,
            },
        },
    },
    tei: {},
}

export const jobOverviewState = {
    jobOverview: {
        activeTypes: [
            'DATAVALUE_IMPORT',
            'EVENT_IMPORT',
            'GEOJSON_IMPORT',
            'GML_IMPORT',
            'METADATA_IMPORT',
        ],
        selectedJob: {
            id: 'EcJpZjEcMfI',
            level: 'INFO',
            created: new Date('2020-02-12T13:00:23.350'),
            lastUpdated: new Date('2020-02-12T13:00:23.766'),
            completed: true,
            events: [
                {
                    id: 'init',
                    text: 'Initiated dataValueImport',
                    date: new Date('2020-02-12T12:00:23.230'),
                },
                {
                    id: 'LPCnEwppbqy',
                    level: 'INFO',
                    text: 'Process started',
                    date: new Date('2020-02-12T12:00:23.232'),
                },
                {
                    id: 'quL1PHwVPg1',
                    level: 'INFO',
                    text: 'Importing data values',
                    date: new Date('2020-02-12T12:00:23.498'),
                },
                {
                    id: 'fn5gnxqYk1J',
                    level: 'INFO',
                    text: 'Import done',
                    date: new Date('2020-02-12T12:00:23.499'),
                },
            ],
            summary: {
                responseType: 'ImportSummary',
                status: 'SUCCESS',
                importOptions: {
                    idSchemes: {},
                    dryRun: true,
                    preheatCache: true,
                    async: true,
                    importStrategy: 'NEW_AND_UPDATES',
                    mergeMode: 'REPLACE',
                    reportMode: 'FULL',
                    skipExistingCheck: false,
                    sharing: false,
                    skipNotifications: false,
                    skipAudit: false,
                    datasetAllowsPeriods: false,
                    strictPeriods: false,
                    strictDataElements: false,
                    strictCategoryOptionCombos: false,
                    strictAttributeOptionCombos: false,
                    strictOrganisationUnits: false,
                    requireCategoryOptionCombo: false,
                    requireAttributeOptionCombo: false,
                    skipPatternValidation: false,
                    ignoreEmptyCollection: false,
                    force: false,
                    firstRowIsHeader: true,
                    skipLastUpdated: false,
                },
                description: 'Import process completed successfully',
                importCount: {
                    imported: 0,
                    updated: 0,
                    ignored: 0,
                    deleted: 0,
                },
                dataSetComplete: 'false',
            },
            error: false,
            importType: 'DATAVALUE_IMPORT',
            jobDetails: {
                files: [],
                format: {
                    value: 'json',
                    label: 'JSON',
                },
                dryRun: true,
                strategy: {
                    value: 'NEW_AND_UPDATES',
                    label: 'New and updates',
                },
                preheatCache: true,
                skipAudit: false,
                dataElementIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                orgUnitIdScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                idScheme: {
                    value: 'UID',
                    label: 'Uid',
                },
                skipExistingCheck: false,
                firstRowIsHeader: false,
            },
        },
    },
}

export const allJobs = ['data', 'event', 'geojson', 'gml', 'metadata', 'tei']
    .map((type) =>
        Object.keys(tasksState[type]).map((id) => tasksState[type][id])
    )
    .flat()
