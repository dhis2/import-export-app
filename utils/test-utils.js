import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
