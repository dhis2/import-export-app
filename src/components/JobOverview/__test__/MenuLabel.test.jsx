import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { MenuLabel } from '../MenuLabel/MenuLabel.jsx'
import { allJobs } from './data.js'

it('matches snapshot', () => {
    const { asFragment } = render(<MenuLabel task={allJobs[0]} />)
    expect(asFragment()).toMatchSnapshot()
})
