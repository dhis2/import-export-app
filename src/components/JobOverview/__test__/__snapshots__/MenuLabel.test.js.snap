// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="job-overview-menu-label-EcJpZjEcMfI"
  >
    <div
      class="filename"
    >
      data1.json
    </div>
    <div
      class="subtitle"
    >
      <div
        class="jsx-72316749 positive"
        data-test="dhis2-uicore-tag"
      >
        <span
          class="jsx-1110058759"
          data-test="dhis2-uicore-tag-text"
        >
          Completed
        </span>
      </div>
       
      <p
        class="date"
      >
        2020-02-12 13:00:23

      </p>
    </div>
  </div>
</DocumentFragment>
`;
